blog/2019-09-ngrx-data-views/README.md

```diff
@@ -0,0 +1,419 @@
+---
+title: "Handling normalized entities with data views"
+author: Danilo Hoffmann
+mail: dhhyi@aol.com
+published: 2019-09-30
+last-change: 2019-09-30
+keywords:
+  - Angular
+  - RxJS
+  - Reactive Programming
+  - Observables
+  - NgRx
+  - NgRx Entities
+  - Memoization
+language: en
+thumbnail: data-views-header.jpg
+hidden: false
+---
+
+Dealing with data supplied by enterprise systems can be quite tedious. Most of the time, data for a single entity has to be retrieved with multiple calls, because it is handled in a normalized way on the server. This Normalization however does a good job minimizing the space used up in any database by preventing duplication and with that also inconsistency. Have a look at the [Wikipedia Article on Database Normalization](https://en.wikipedia.org/wiki/Database_normalization) for more information.
+
+If the API giving us access to the data is just a simple REST API exposing all entities via separate endpoints, we have to accumulate the needed data in the front-end ourselves.
+
+## Data Clustering in Enterprise Applications
+
+Have a look at the [WordPress API Reference](https://developer.wordpress.org/rest-api/reference/) for such an example. Individual Posts can be retrieved quite typically via `/posts/<id>`. However, the returned data just contains IDs for the author, tags and categories of the post. To display a post properly this additional data has to be retrieved with subsequent calls to other endpoints.
+
+The most effective way to handle this clustering in this example is to fetch all data for categories and tags with list calls when the application is starting up. This data can then be cached and referenced whenever needed. `@ngrx/entity` provides a perfect API for handling multiple entities in Angular applications. The data can be stored normalized exactly the way it is retrieved via the API.
```

Comment:
Mention of ngrx/entity is too fast IMO. There should be a bridging phrase like "Let's assume we have an Angular application with NgRx for state management. This way, ngrx/entity provides a perfect"

and later:

"If you are not familiar with NgRx or `@ngrx/entity` read here, here and here"

Reply:
I tried, maybe a bit too condescending. Still ok? :sweat_smile: